analyzer/src/main/kotlin/managers/Carthage.kt

```diff
@@ -0,0 +1,282 @@
+            ),
+            vcs = vcsInfo
+        )
+    }
+
+    private fun createPackageFromGithubUserAndProject(id: String, ref: String): Package {
+        val projectUrl = "https://github.com/$id"
+        val vcsInfo = VcsInfo(
+            type = VcsType.GIT,
+            url = "$projectUrl.git",
+            revision = ref
+        )
+        return Package(
+            id = Identifier(
+                type = managerName,
+                namespace = VcsHost.GITHUB.getUserOrOrganization(projectUrl).orEmpty(),
+                name = VcsHost.GITHUB.getProject(projectUrl).orEmpty(),
+                version = ref
+            ),
+            declaredLicenses = sortedSetOf(),
+            description = "",
+            homepageUrl = projectUrl,
+            binaryArtifact = RemoteArtifact.EMPTY,
+            sourceArtifact = RemoteArtifact(
+                url = VcsHost.GITHUB.getArchiveDownloadUrl(vcsInfo).orEmpty(),
+                hash = Hash.NONE
+            ),
+            vcs = vcsInfo
+        )
+    }
+
+    private fun createPackageFromLocalRepository(fileUrl: String, ref: String): Package {
+        val vcsInfo = VcsInfo(
+            type = VcsType.GIT,
+            url = fileUrl,
+            revision = ref
+        )
+        return Package(
+            id = Identifier(
+                type = managerName,
+                namespace = "",
+                name = fileUrl.substringAfterLast("/"),
+                version = ref
+            ),
+            declaredLicenses = sortedSetOf(),
+            description = "",
+            homepageUrl = "",
+            binaryArtifact = RemoteArtifact.EMPTY,
+            sourceArtifact = RemoteArtifact.EMPTY,
+            vcs = vcsInfo
+        )
+    }
+
+    private fun createPackageFromBinarySpec(binarySpec: Map<String, String>, id: String, ref: String) = Package(
+        id = Identifier(
+            type = managerName,
+            namespace = "",
+            name = id.substringAfterLast("/").removeSuffix(".json"),
+            version = ref
+        ),
+        declaredLicenses = sortedSetOf(),
+        description = "",
+        homepageUrl = "",
+
+        // The content from the artifactUrl might contain header files that should be scanned.
+        // Therefore adding this URL for both sourceArtifact and binaryArtifact.
+        binaryArtifact = RemoteArtifact(
+            url = binarySpec[ref].orEmpty(),
+            hash = Hash.NONE
+        ),
+        sourceArtifact = RemoteArtifact(
+            url = binarySpec[ref].orEmpty(),
+            hash = Hash.NONE
+        ),
+        vcs = VcsInfo.EMPTY
+    )
+
+    private fun isGithubUserAndProject(id: String): Boolean {
+        val regex = Regex("([a-zA-Z0-9]+)/([a-zA-Z0-9]+)")
+        val match = regex.matchEntire(id)
+        return match != null
+    }
+
+    private fun isAbsolutePath(filePath: String) = filePath.startsWith("/")
+
+    private fun isFilePath(workingDir: String, id: String) =
```

Comment:
Why is the variable named id?

Reply:
Because Carthage calls it `id`. But you're right, in this context `id` doesn't make sense.
Renamed it to `path` but keeping the calling variable name `id`.